### PR TITLE
put all network checking logic in check_fix_network.sh

### PR DESCRIPTION
**Pull request by jimyang2008**

### CODE REVIEW

**Comment by jerrchen** [comment]
master is frozen until we push to release_4, likely be open Wed (Pacific Time)

**Comment by MelvinTo** [comment]
Overall, looks a little complex. too many "goto" like statements. No need to change though.

**Comment by MelvinTo** [comment]
see my code review comments for some minor issues to fix @jimyang2008

File: scripts/check_fix_network.sh [hunk 1]
```diff
@@ -19,32 +19,35 @@
...
+    for kind in ip gw
     do
         value=$(get_value $kind)
         [[ -n "$value" ]] || continue
         file=/var/run/saved_${kind}
-        $LOGGER "Save $kind value $value in $file"
         rm -f $file
         echo "$value" > $file || r=1
     done
+    [[ -e /etc/resolv.conf.bak ]] || /bin/cp /etc/resolv.conf{,.bak}
```
[the remark this answers is not in the document]
Comment (jimyang2008) [reply scripts/check_fix_network.sh]: Sure. I will change it to override the previous backup, as long as `save_values` is only called at working state.

**Pull request merged**